### PR TITLE
Some code cleanups tagged by static analysis

### PR DESCRIPTION
**Pull request by seelabs**
## High Level Overview of Change

Misc code cleanups tagged by static analysis. This include reducing copies, removing unneeded moves, and adding explicit.

### Type of Change

- [x ] Refactor (non-breaking change that only restructures code)

### CODE REVIEW

File: src/ripple/app/rdb/backend/RelationalDBInterfaceSqlite.cpp [hunk 1]
```diff
@@ -174,13 +174,13 @@ class RelationalDBInterfaceSqliteImp : public RelationalDBInterfaceSqlite
...
+    std::uint64_t
     getKBUsedAll() override;
 
-    int
+    std::uint64_t
     getKBUsedLedger() override;
 
-    int
+    std::uint64_t
     getKBUsedTransaction() override;
```
Comment (HowardHinnant) [inline src/ripple/app/rdb/backend/RelationalDBInterfaceSqlite.cpp:185]: The use of these three functions in src/ripple/rpc/handlers/GetCounts.cpp needs updating to reflect the change from `int` to `std::uint64_t`.  Also in src/ripple/core/SociDB.h we're using `size_t` for these returns.  We should make these the same type alias.

**Review by HowardHinnant — CHANGES_REQUESTED** [review]
Looks good.  Just one minor comment that needs a small fix.

File: src/ripple/app/rdb/backend/RelationalDBInterfaceSqlite.cpp (hunk 1 above)
Comment (seelabs) [reply src/ripple/app/rdb/backend/RelationalDBInterfaceSqlite.cpp:185]: Thanks! I changed them all to `std::uint32_t`. Fixed in `975aebe35c [fold] Measure db memory usage with a uint32 var
`

**Review by HowardHinnant — APPROVED** [review]

**Review by intelliot — APPROVED** [review]
I'm curious whether we expect this to have any impact on performance test results?

**Comment by seelabs** [comment]
@intelliot While these changes do reduce allocations, I doubt it's measurable in anything but a micro-benchmark.

**Pull request closed without merging**